TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1156,6 +1159,283 @@ def exterior_boundaries(feature_layers, zoom,
+    """
+    keys = set(lprops.keys()) | set(rprops.keys())
+    new_props = dict()
+
+    # props in both are copied directly if they're the same
+    # in both the left and right. they get left/right
+    # inserted after the first ':' if they're different.
+    for k in keys:
+        lv = lprops.get(k)
+        rv = rprops.get(k)
+
+        if lv == rv:
+            new_props[k] = lv
+        else:
+            if lv is not None:
+                new_props[_inject_key(k, 'left')] = lv
+            if rv is not None:
+                new_props[_inject_key(k, 'right')] = rv
+
+    return new_props
+
+
+def _make_joined_name(props):
+    """
+    Updates the argument to contain a 'name' element
+    generated from joining the left and right names.
+
+    Just to make it easier for people, we generate a name
+    which is easy to display of the form "LEFT - RIGHT".
+    The individual properties are available if the user
+    wants to generate a more complex name.
+
+    >>> x = {}
+    >>> _make_joined_name(x)
+    >>> x
+    {}
+
+    >>> x = {'name:left':'Left'}
+    >>> _make_joined_name(x)
+    >>> x
+    {'name': 'Left', 'name:left': 'Left'}
+
+    >>> x = {'name:right':'Right'}
+    >>> _make_joined_name(x)
+    >>> x
+    {'name': 'Right', 'name:right': 'Right'}
+
+    >>> x = {'name:left':'Left', 'name:right':'Right'}
+    >>> _make_joined_name(x)
+    >>> x
+    {'name:right': 'Right', 'name': 'Left - Right', 'name:left': 'Left'}
+
+    >>> x = {'name:left':'Left', 'name:right':'Right', 'name': 'Already Exists'}
+    >>> _make_joined_name(x)
+    >>> x
+    {'name:right': 'Right', 'name': 'Already Exists', 'name:left': 'Left'}
+    """
+
+    # don't overwrite an existing name
+    if 'name' in props:
+        return
+
+    lname = props.get('name:left')
+    rname = props.get('name:right')
+
+    if lname is not None:
+        if rname is not None:
+            props['name'] = "%s - %s" % (lname, rname)
+        else:
+            props['name'] = lname
+    elif rname is not None:
+        props['name'] = rname
+
+
+def _linemerge(geom):
+    """
+    Try to extract all the linear features from the geometry argument
+    and merge them all together into the smallest set of linestrings
+    possible.
+
+    This is almost identical to Shapely's linemerge, and uses it,
+    except that Shapely's throws exceptions when passed a single
+    linestring, or a geometry collection with lines and points in it.
+    So this can be thought of as a "safer" wrapper around Shapely's
+    function.
+    """
+    geom_type = geom.type
+
+    if geom_type == 'GeometryCollection':
+        # collect together everything line-like from the
+        # geometry collection
+        lines = map(_linemerge, geom.geoms)
+
+        # filter out anything that's empty
+        lines = filter(lambda g: not g.is_empty, lines)
+
+        if len(lines) == 0:
+            return MultiLineString([])
+        else:
+            return linemerge(lines)
```

Comment:
Is it ok if we make this less functional and more imperative? I think that's generally the more preferred style in python. Something like:

```python
lines = []
for subgeom in geom.geoms:
    line = _linemerge(subgeom)
    if not line.is_empty:
        lines.append(subline)
return linemerge(lines) if lines else MultiLineString([])
```